src/components/ToDoList/ToDoList.test.tsx

```diff
@@ -42,79 +51,99 @@ describe("ToDoList", () => {
+            expect(screen.queryByLabelText(deleteButtonSelector)).not.toBeInTheDocument()
         });
     });
 
     it("should display the number of to-do items", async () => {
         // Arrange
-        const toDoItems = [faker.lorem.word(), faker.lorem.word()];
-        jest.spyOn(toDoService, "getToDoList").mockReturnValue(toDoItems);
+        const toDoItems = createToDoItems(2);
+        const expectedNewItem = createToDoItems(1)[0];
+        jest.spyOn(toDoService, "getToDoList")
+            .mockImplementationOnce(() => Promise.resolve(toDoItems)) // initial load
+            .mockImplementationOnce(() => Promise.resolve([...toDoItems, expectedNewItem])); 
+        jest.spyOn(toDoService, "addToDoItem").mockImplementation(() => Promise.resolve(new Response()));
+
+        // Act
         render(<ToDoList />);
 
-        expect(screen.getByText("Number of To-Do List items: 2")).toBeInTheDocument();
+        // Assert
+        await waitFor(async () => {
+            expect(await screen.findByText("Number of To-Do List items: 2")).toBeInTheDocument();
+        });
 
         // Act
-        const newToDoItem = faker.lorem.word();
-        const input = screen.getByLabelText("todo-input");
-        userEvent.paste(input, newToDoItem);
+        const input = await screen.findByLabelText("todo-input");
+        userEvent.paste(input, expectedNewItem.description);
         userEvent.click(screen.getByRole("button", { name: "Add To-Do Item" }));
 
         // Assert
-        await waitFor(() => {
-            expect(screen.getByText("Number of To-Do List items: 3")).toBeInTheDocument();
+        await waitFor(async () => {
+            expect(await screen.findByText("Number of To-Do List items: 3")).toBeInTheDocument();
         });
     });
 
     it("should display a message when the to-do list is complete", async () => {
         // Arrange
         const expectedText = "You did everything on your list!";
-        const itemToDelete = faker.lorem.word();
-        jest.spyOn(toDoService, "getToDoList").mockReturnValue([itemToDelete]);
-        render(<ToDoList />);
+        const toDoItems = createToDoItems(1);
+        jest.spyOn(toDoService, "getToDoList")
+            .mockImplementationOnce(() => Promise.resolve(toDoItems)) // initial load
+            .mockImplementationOnce(() => Promise.resolve([])); // after delete
+        jest.spyOn(toDoService, "deleteToDoItem").mockImplementation(() => Promise.resolve(new Response()));
 
         // Act
-        userEvent.click(screen.getByLabelText(`delete-${itemToDelete}-0`));
+        render(<ToDoList />);
+        userEvent.click(await screen.findByLabelText(`delete-${toDoItems[0].description}-${toDoItems[0].id}`));
 
         // Assert
-        await waitFor(() => {
-            expect(screen.getByText(expectedText)).toBeInTheDocument();
+        await waitFor(async () => {
+            expect(await screen.findByText(expectedText)).toBeInTheDocument();
```

Comment:
just `await findBy` is all we should need in cases like this

Reply:
I think this was left over from troubleshooting. I'll clean these up. Thanks!